### PR TITLE
Skymap display

### PR DESCRIPTION
**Pull request by mcoughlin**
This PR merges the skymap display and form into one component.

### CODE REVIEW

**Review by stefanv — APPROVED** [review]

File: static/js/components/GeoJSONGlobePlot.jsx [hunk 1]
```diff
@@ -46,7 +68,11 @@ const GeoJSONGlobePlot = ({ skymap, sources, galaxies, instrument }) => {
     // eslint-disable-next-line
     galaxies,
     // eslint-disable-next-line
-    instrument
+    instrument,
+    // eslint-disable-next-line
+    observations,
+    // eslint-disable-next-line
+    options
```
Comment (guynir42) [inline static/js/components/GeoJSONGlobePlot.jsx:75]: Just curious, what does eslint not like about these inputs?

File: static/js/components/LocalizationPlot.jsx [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+  const localization =
+    loc.id === cachedLocalization?.id ? cachedLocalization : null;
+
+  if (!localization || !instrument) {
+    return <CircularProgress />;
+  }
+
+  return (
+    <>
+      <GeoJSONGlobePlot
```
Comment (guynir42) [inline static/js/components/LocalizationPlot.jsx:36]: Is it possible to get rid of `LocalizationPlot` and just put the code from this component into `GeoJSONGlobePlot`?

File: static/js/components/GcnSelectionForm.jsx [hunk 3]
```diff
@@ -62,14 +62,18 @@ const GcnSelectionForm = ({ gcnEvent }) => {
     "instrument",
     "observations",
   ];
+  const displayOptionsDefault = displayOptions.reduce(
```
Comment (stefanv) [inline static/js/components/GcnSelectionForm.jsx:65]: This is cute, but not very readable. Perhaps consider:

```
Object.fromEntries(displayOptions.map(x => [x, false]))
```

File: static/js/components/LocalizationPlot.jsx [hunk 4]
```diff
@@ -136,11 +139,259 @@ LocalizationPlot.propTypes = {
       processed_fraction: PropTypes.number,
     })
   ).isRequired,
-  options: PropTypes.arrayOf(PropTypes.number),
+  options: PropTypes.arrayOf(PropTypes.number).isRequired,
 };
 
-LocalizationPlot.defaultProps = {
-  options: [false, false, false, false, false],
+const useD3 = (
```
Comment (stefanv) [inline static/js/components/LocalizationPlot.jsx:145]: This feels like an abuse of `useD3`.  How about signature:

```
useD3 = (renderer, height, width, data)
```

instead, where `data` can contain anything you want. The idea is that `useD3` could be used more widely than just this module, but that then the renderer and the input data would change.

**Review by stefanv — COMMENTED** [review]
This looks good overall; please fix the react PropType errors.

File: static/js/components/GeoJSONGlobePlot.jsx (hunk 1 above)
Comment (mcoughlin) [reply static/js/components/GeoJSONGlobePlot.jsx]: Got rid of this.

File: static/js/components/GeoJSONGlobePlot.jsx [hunk 5]
```diff
@@ -92,7 +118,7 @@ const GeoJSONGlobePlot = ({ skymap, sources, galaxies, instrument }) => {
         .style("stroke-width", "0.5px")
         .attr("d", geoGenerator);
 
-      if (skymap?.features) {
+      if (skymap?.features && options[0]) {
```
[the remark this answers is not in the document]
Comment (mcoughlin) [reply static/js/components/GeoJSONGlobePlot.jsx]: Yeah this was a good idea. Implemented.

File: static/js/components/LocalizationPlot.jsx (hunk 2 above)
Comment (mcoughlin) [reply static/js/components/LocalizationPlot.jsx:39]: MErged.

File: static/js/components/GcnSelectionForm.jsx (hunk 3 above)
Comment (mcoughlin) [reply static/js/components/GcnSelectionForm.jsx]: Yup added.

File: static/js/components/LocalizationPlot.jsx (hunk 4 above)
Comment (mcoughlin) [reply static/js/components/LocalizationPlot.jsx]: Yes this is a nice idea.

**Comment by stefanv** [comment]
Looking good!

Only one error remaining:

```
react.development.js?72d0:220 Warning: Failed prop type: Invalid GeoJSON for GeoJSONGlobePlot At 0: must have a member with the name "properties".
```

**Comment by stefanv** [comment]
The following 2MB request is made three times in a row:

```
http://localhost:5000/api/instrument?localizationDateobs=2019-04-25T08%3A18%3A05&includeGeoJSONSummary=true
http://localhost:5000/api/instrument?localizationDateobs=2019-04-25T08%3A18%3A05&includeGeoJSONSummary=true
http://localhost:5000/api/instrument?localizationDateobs=2019-04-25T08%3A18%3A05&includeGeoJSONSummary=true
```

May want to make use of Redux caching.

**Comment by stefanv** [comment]
It's oh, so green!

**Pull request merged**